src/main/java/org/testng/reporters/FailedReporter.java

```diff
@@ -143,25 +110,31 @@ private void generateXmlTest(ISuite suite,
       // sorted, we don't need to sort them again.
       //
       List<ITestNGMethod> result = Lists.newArrayList();
+      Set<ITestNGMethod> relevantConfigs = Sets.newHashSet();
```

Comment:
Why using a temp set? Is it to avoid duplication?
Maybe we can replace `List result` by `LinkedHashSet result` to have the order respected and avoid duplication.